pinot-common/src/main/java/org/apache/pinot/common/utils/CommonConstants.java

```diff
@@ -197,6 +197,9 @@ public ServerType getServerType() {
     public static final String CONFIG_OF_ENABLE_COMMIT_END_WITH_METADATA =
         "pinot.server.instance.enable.commitend.metadata";
     public static final String CONFIG_OF_REALTIME_OFFHEAP_ALLOCATION = "pinot.server.instance.realtime.alloc.offheap";
+    public static final String CONFIG_OF_REALTIME_ENABLE_UPLOAD_TO_CONTROLLER = "pinot.server.instance.enable.upload.segment.to.controller";
```

Comment:
what does this enable do? What is the behavior when it is not enabled?

Reply:
This parameter controls if a committer server uploads segments to the controller or not. It should be named to enableSegmentUploadToController(). If it is not enabled, the server will only upload meta data the controller.

Reply:
Comments added.

Reply:
We have re-factored the code to have a separate committer class. Please re-factor it further to allow the code to use any committer class. The committer class should be configured in table config. You can then add all the logic on the server side without needing this config